src/h3lib/lib/faceijk.c

```diff
@@ -362,35 +362,74 @@ static const int unitScaleByCIIres[] = {
 };
 
 /**
- * Encodes a coordinate on the sphere to the FaceIJK address of the containing
- * cell at the specified resolution.
+ * Encodes a Vec3d coordinate to the corresponding icosahedral face and
+ * squared euclidean distance to that face center.
  *
- * @param g The spherical coordinates to encode.
- * @param res The desired H3 resolution for the encoding.
- * @param h The FaceIJK address of the containing cell at resolution res.
+ * @param v3d The Vec3d coordinates to encode.
+ * @param face The icosahedral face containing the coordinates.
+ * @param sqd The squared euclidean distance to its icosahedral face center.
  */
-void _geoToFaceIjk(const LatLng *g, int res, FaceIJK *h) {
-    // first convert to hex2d
-    Vec2d v;
-    _geoToHex2d(g, res, &h->face, &v);
+static void _vec3dToClosestFace(const Vec3d *v3d, int *face, double *sqd) {
+    // determine the icosahedron face
+    *face = 0;
+    // The distance between two farthest points is 2.0, therefore the square of
+    // the distance between two points should always be less or equal than 4.0 .
+    *sqd = 5.0;
+    for (int f = 0; f < NUM_ICOSA_FACES; ++f) {
+        double sqdT = vec3DistSq(&faceCenterPoint[f], v3d);
+        if (sqdT < *sqd) {
+            *face = f;
+            *sqd = sqdT;
+        }
+    }
+}
```

Comment:
This code is pretty much the same as what we currently have, Github is showing a more complicated diff because the functions were reordered in the file.